pandas/core/frame.py

```diff
@@ -438,7 +438,7 @@ def __init__(
         elif isinstance(data, dict):
             mgr = init_dict(data, index, columns, dtype=dtype)
         elif isinstance(data, ma.MaskedArray):
-            import numpy.ma.mrecords as mrecords
+            from numpy.ma import mrecords
```

Comment:
Hmm actually now that I think about it is this an isort thing or something you are suggesting as a standard? 

These aren't entirely equivalent and this change is more prone to circular imports. The former is actually suggest by Guido as the way to go so I don't think this is worth changing. (can't find link atm, but its somewhere in the Python docs)

Reply:
This is in fact an ```isort``` thing (in the dev version), If you can find the link I think we should open an issue about this (to ```isort```).

The first commit is the changes made by ```isort```, the second commit was done manually to fix the pattern of:

```python
from foo.bar import baz as baz
```

Reply:
`import numpy.ma` doesnt import `mrecords` into the `np.ma` namespace.  Is this pattern ever liable to trip us up with either form of import?

Reply:
@MomIsBestFriend here is what I was describing:

https://docs.python.org/3/faq/programming.html#how-can-i-have-modules-that-mutually-import-each-other

> Guido van Rossum recommends avoiding all uses of from <module> import ..., and placing all code inside functions.

So while this is inside a function, I think still not worth changing to a usage that is discouraged by Guido himself

Reply:
Thank you @WillAyd for sending the link!

I have opened an issue about this discussion at ```isort```. 

---

xref https://github.com/timothycrosley/isort/issues/1164

Reply:
Regardless of the `from x import y` conversation, these usages are not needed since you are not renaming:

`import numpy.na.mrecords` is the same as `import numpy.na.mrecords as mrecords` (will probably help you at least workaround this decision by `isort` team).